src/nrniv/nrncore_write/callbacks/nrncore_callbacks.cpp

```diff
@@ -577,6 +635,8 @@ int* datum2int(int type,
             } else if (etype == -7) {  // bbcorepointer
                 pdata[jj] = ml_vdata_offset + eindex;
                 // printf("etype %d jj=%d eindex=%d pdata=%d\n", etype, jj, eindex, pdata[jj]);
+            } else if (etype == -11) {  // random
```

Comment:
this whole piece of code could be a `switch case` - again, a nitpick and not necessarily relevant to this PR.

Reply:
Deferring to separate PR or when ```NEURON { VECTOR vec1, ...}``` is introduced.